src/py/flwr/server/driver/driver.py

```diff
@@ -69,44 +70,185 @@ def _get_grpc_driver_and_run_id(self) -> Tuple[GrpcDriver, int]:
-
         # Call GrpcDriver method
         res = grpc_driver.get_nodes(GetNodesRequest(run_id=run_id))
-        return list(res.nodes)
+        return [node.node_id for node in res.nodes]
 
-    def push_task_ins(self, task_ins_list: List[TaskIns]) -> List[str]:
-        """Schedule tasks."""
-        grpc_driver, run_id = self._get_grpc_driver_and_run_id()
+    def push_messages(self, messages: Iterable[Message]) -> Iterable[str]:
+        """Push messages to specified node IDs.
+
+        This method takes an iterable of messages and sends each message
+        to the node it specifies in `node_id`.
 
-        # Set run_id
-        for task_ins in task_ins_list:
-            task_ins.run_id = run_id
+        Parameters
+        ----------
+        messages : Iterable[Message]
+            An iterable of messages to be sent.
 
+        Returns
+        -------
+        message_ids : Iterable[str]
+            An iterable of IDs for the messages that were sent, which can be used
+            to pull responses.
+        """
+        grpc_driver, _ = self._get_grpc_driver_and_run_id()
+        # Construct TaskIns
+        task_ins_list: List[TaskIns] = []
+        for msg in messages:
+            # Check message
+            self._check_message(msg)
+            # Convert Message to TaskIns
+            taskins = message_to_taskins(msg)
+            # Add to list
+            task_ins_list.append(taskins)
         # Call GrpcDriver method
         res = grpc_driver.push_task_ins(PushTaskInsRequest(task_ins_list=task_ins_list))
         return list(res.task_ids)
 
-    def pull_task_res(self, task_ids: Iterable[str]) -> List[TaskRes]:
-        """Get task results."""
-        grpc_driver, _ = self._get_grpc_driver_and_run_id()
+    def pull_messages(self, message_ids: Iterable[str]) -> Iterable[Message]:
+        """Pull messages based on message IDs.
 
-        # Call GrpcDriver method
+        This method is used to collect responses or messages from the network
+        that correspond to a set of given task identifiers.
+
+        Parameters
+        ----------
+        message_ids : Iterable[str]
+            An iterable of task identifiers for which messages are to be retrieved.
+
+        Returns
+        -------
+        Iterable[Message]
+            An iterable of messages received.
+        """
+        grpc_driver, _ = self._get_grpc_driver_and_run_id()
+        # Pull TaskRes
         res = grpc_driver.pull_task_res(
-            PullTaskResRequest(node=self.node, task_ids=task_ids)
+            PullTaskResRequest(node=self.node, task_ids=message_ids)
         )
-        return list(res.task_res_list)
+        # Convert TaskRes to Message
+        msgs = [message_from_taskres(taskres) for taskres in res.task_res_list]
+        return msgs
+
+    def send_and_receive(
+        self,
+        messages: Iterable[Message],
+        *,
+        time_out: Optional[float] = None,
```

Comment:
There's no `_` in other instances of `timeout` in the codebase:
```suggestion
        timeout: Optional[float] = None,
```